extensions/ql-vscode/src/query-history-serializater.ts

```diff
@@ -0,0 +1,178 @@
+import { pathExists, readFile, remove, mkdir, writeFile } from "fs-extra";
+import { dirname } from "path";
+
+import { showAndLogErrorMessage } from "./helpers";
+import {
+  asyncFilter,
+  getErrorMessage,
+  getErrorStack,
+} from "./pure/helpers-pure";
+import { CompletedQueryInfo, LocalQueryInfo } from "./query-results";
+import { QueryHistoryInfo } from "./query-history/query-history-info";
+import { QueryStatus } from "./query-status";
+import { QueryEvaluationInfo } from "./run-queries-shared";
+import { QueryResultType } from "./pure/legacy-messages";
+import { RemoteQueryHistoryItem } from "./data-serialization/source-schemas-types/remote-query-history-item";
+import { VariantAnalysisHistoryItem } from "./data-serialization/source-schemas-types/variant-analysis-history-item";
+import { App } from "./common/app";
+import { DisposableObject } from "./pure/disposable-object";
+
+type SerializableHistoryItems =
+  | LocalQueryInfo
+  | RemoteQueryHistoryItem
+  | VariantAnalysisHistoryItem;
+
+export class QueryHistorySerializer extends DisposableObject {
+  private readonly fsPath: string;
+  private readonly jsonValidator: App["jsonValidator"];
```

Comment:
Any reason not to do this?

```suggestion
  private readonly jsonValidator: JsonValidator;
```